build-common/src/org/jetbrains/kotlin/incremental/storage/LookupMap.kt

```diff
@@ -35,23 +35,13 @@ class LookupMap(
     override fun dumpValue(value: Collection<Int>): String = value.toString()
 
     fun add(name: String, scope: String, fileId: Int) {
-        storage.append(LookupSymbolKey(name, scope), listOf(fileId))
+        storage.append(LookupSymbolKey(name, scope), fileId)
     }
 
-    fun append(lookup: LookupSymbolKey, fileIds: Collection<Int>) {
-        storage.append(lookup, fileIds)
-    }
-
-    operator fun get(key: LookupSymbolKey): Collection<Int>? = storage[key]
+    override fun get(key: LookupSymbolKey): Collection<Int>? = storage[key]?.toSet()
```

Comment:
What do you think about making the map values as `Set` in the first place? Now it's declared to be a map of `LookupSymbolKey -> Collection<Int>`, however effectively it is `LookupSymbolKey -> Set<Int>`, we deserialize the values as `List` and then convert the list into a `Set`. Perhaps we could avoid that conversion. On the other side, such a change will make the append logic of `InMemoryStorage` more complex

Reply:
Yup you’re right. I plan to declare this map as `LookupSymbolKey -> Set<Int>` in the follow-up PR. (This PR focuses on the abstract classes; the next PR will clean up the subclasses – e.g., removing all these overriding methods in the subclasses.)